### PR TITLE
Api error

### LINKED ISSUE #13: ApiError
There must be only one error type in server, which is ApiError. Therefore, HeaderValidationError must be substituted by ApiError.

### PR DESCRIPTION
**Pull request by danielmoraes**
Depends on #15. Closes #13.

### CODE REVIEW

File: server/errors/api-error.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * @extends AbstractError
+ */
+class ApiError extends AbstractError {
+  /**
+   * Creates an API error.
+   * @param {string} message - Error message.
+   * @param {number} status - HTTP status code.
+   * @param {boolean} isPublic - Whether the error message should be public.
+   */
+  constructor (message, status = httpStatus.INTERNAL_SERVER_ERROR,
```
Comment (danielmoraes) [inline server/errors/api-error.js]: It is not the opposite? If the arguments with default values are in the end, we can instantiate `ApiError` as:
`new ApiError('message')`

Comment (reinert) [inline server/errors/api-error.js]: Sure! You're right! Sorry! One thing came to my mind... 500 family will not be thrown by ApiError, only the 400 family. So I think 400 (BAD_REQUEST) should be the default status error.

INTERNAL_SERVER_ERROR will occur unexpectedly. This is handled by the uncaught error handler.

File: server/index.js [hunk 2]
```diff
@@ -1,14 +1,18 @@
...
 
 module.exports = express()
   .use(morgan(process.env.NODE_ENV === 'production' ? 'combined' : 'dev'))
   .use(helmet())
   .use(bodyParser.json({ type: 'application/json' }))
+  .use('/auth', authRouter)
   .use('/users', userRouter)
-  .use(ErrorHandler.all())
-  .get('/', (req, res) => res.send('Hello World! \\o/'))
+  .use((req, res, next) =>
```
Comment (reinert) [inline server/index.js]: Declare this function below and just reference it here.

File: server/errors/api-error.js (hunk 1 above)
Comment (reinert) [inline server/errors/api-error.js]: Also, I don't see any use for the `isPublic` argument.

File: server/handlers/auth-handler.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      })
+      .then(user => user.comparePassword(req.password))
+      .then(success => {
+        if (!success) {
+          const message = 'Invalid credentials'
+          return next(new ApiError(message, httpStatus.UNAUTHORIZED, true))
+        }
+
+        const secret = process.env.JWT_SECRET
+        const options = { expiresIn: '1d' }
```
Comment (reinert) [inline server/handlers/auth-handler.js]: Shouldn't these options be located at /config/jwt.json ?

**Pull request merged**